database/database.go

```diff
@@ -0,0 +1,151 @@
+package database
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+
+	"github.com/HouzuoGuo/tiedot/db"
+	"github.com/fatih/structs"
+	"github.com/josephroberts/edge-node-manager/config"
+	"github.com/josephroberts/edge-node-manager/device"
+)
+
+/*
+Uses the tiedot database
+https://github.com/HouzuoGuo/tiedot
+*/
+
+var directory string
+var connection *db.DB
+
+func LoadDevices(uuid string, t *device.Type) (map[int]device.Interface, error) {
+	initialise()
+	query := fmt.Sprintf(`[{"eq": "%s", "in": ["ApplicationUUID"]}]`, uuid)
```

Comment:
It seems like a good idea to break out this logic into a helper function. As this `LoadDevices()` function is full of confusing stuff relating to how tiedb does queries rather than how to load devices rendering it a lot less readable.